### PR TITLE
Introduce SafeLog method for errors

### PR DESCRIPTION
**Pull request by colin-higgins**
@DataDog/apm-dotnet

### CODE REVIEW

**Comment by zacharycmontoya** [comment]
I'm curious, does this solve any one issue in particular, or is this just general exception handling goodness?

File: src/Datadog.Trace/Logging/DatadogLogging.cs [hunk 1]
```diff
@@ -94,6 +94,18 @@ public static ILogger For<T>()
...
 
+        public static void SafeLogError(this ILogger logger, Exception ex, string message, params string[] args)
+        {
+            try
+            {
+                logger.Error(ex, message, args);
+            }
+            catch
+            {
+                // ignore
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Logging/DatadogLogging.cs:105]: Should we at least output to stdout? I have reservations about completely ignoring exceptions.

Comment (colin-higgins) [reply src/Datadog.Trace/Logging/DatadogLogging.cs:105]: Sure, IMO, it's a best effort reporting. 
Number one priority is not to crash apps.

**Comment by colin-higgins** [comment]
@zacharycmontoya General exception handling. It's prophylactic mostly.
My thought is that we should by policy have "best effort" logging in catch blocks. 
Logging should never throw there, so we may as well have a global funnel where we prevent it from bubbling up.

**Pull request merged**